storage/scylla_api.c

```diff
@@ -0,0 +1,302 @@
+      mallocAndCopyText(&(((*transaction) + idx)->message), text, len);
+      cass_value_get_int64(cass_row_get_column_by_name(row, "value"), &val);
+      ((*transaction) + idx)->value = val;
+      cass_value_get_bytes(cass_row_get_column_by_name(row, "timestamp"), &buf, &len);
+      mallocAndCopyBlob(&(((*transaction) + idx)->timestamp), buf, len);
+      idx++;
+    }
+
+    cass_result_free(result);
+    cass_iterator_free(iterator);
+  }
+
+  cass_future_free(future);
+  cass_statement_free(statement);
+
+  return rc;
+}
+
+static int init(CassCluster* cluster, CassSession* session, char argc, char* argv[]) {
+  char* hosts = "172.17.0.2";
+
+  if (argc > 1) {
+    hosts = argv[1];
+  }
+  /* Add contact points */
+  cluster = create_cluster(hosts);
+
+  if (connect_session(session, cluster) != CASS_OK) {
+    cass_cluster_free(cluster);
+    cass_session_free(session);
+    return -1;
+  }
+
+  execute_query(session,
+                "CREATE KEYSPACE IF NOT EXISTS permanent WITH replication = { \
+                                              'class': 'SimpleStrategy', 'replication_factor': '3' };");
+  execute_query(session,
+                "CREATE TABLE IF NOT EXISTS permanent.bundleTable (bundle blob, \
+                                              address blob, \
+                                              transaction blob, \
+                                              message text, \
+                                              value bigint, \
+                                              timestamp varint, \
+                                              PRIMARY KEY (bundle,timestamp,address));");
+  return 0;
+}
+
+int insert_transaction_into_bundleTable(CassSession* session, Transaction transaction) {
+  const CassPrepared* insert_prepared = NULL;
+  const char* insert_query =
+      "INSERT INTO permanent.bundleTable (bundle, address, transaction, message, value, timestamp) VALUES (?, ?, ?, ?, "
+      "?, ?);";
+
+  if (prepare_query(session, insert_query, &insert_prepared) == CASS_OK) {
+    insert_into_bundleTable(session, insert_prepared, &transaction);
+    cass_prepared_free(insert_prepared);
+  } else {
+    printf("insert error\n");
+    return -1;
+  }
+  return 0;
+}
+
+int select_transactions_with_bundle(CassSession* session, const char* bundle) {
+  Transaction* output;
+  int outputNum = 0;
+  const CassPrepared* select_prepared = NULL;
+  const char* select_query = "SELECT * FROM permanent.bundleTable WHERE bundle = ?";
+
+  if (prepare_query(session, select_query, &select_prepared) == CASS_OK) {
+    select_from_bundleTable(session, select_prepared, bundle, &output, &outputNum);
+    printf("row = %d \n", outputNum);
+
+    for (int i = 0; i < outputNum; i++) {
+      printf("%s %s %s %s %ld %s\n", output[i].bundle, output[i].address, output[i].transaction, output[i].message,
+             output[i].value, output[i].timestamp);
+    }
+
+    if (output) {
+      free(output);
```

Comment:
You don't have to check the parameter of `free`. Simply call `free` without checking its value.